instrument-cloudwatch/src/Instrument/CloudWatch.hs

```diff
@@ -282,3 +289,23 @@ httpRetryH = const $ EX.Handler $ \(_ :: Amazonka.HttpException) -> return True
 -- | 'IOException's should be retried
 networkRetryH :: Monad m => a -> EX.Handler m Bool
 networkRetryH = const $ EX.Handler $ \(_ :: EX.IOException) -> return True
+
+-------------------------------------------------------------------------------
+newtype GzippedPutMetricData = GzippedPutMetricData CW.PutMetricData
+  deriving newtype (AWS.ToPath, AWS.ToQuery, AWS.ToHeaders)
+
+instance Amazonka.Types.AWSRequest GzippedPutMetricData where
+  type AWSResponse GzippedPutMetricData = CW.PutMetricDataResponse
+  request overrides x =
+    let req = (AWSRequest.postQuery (overrides CW.defaultService) x :: Amazonka.Types.Request GzippedPutMetricData)
+        !bs = BS.fromStrict (AWS.toBS (AWS.toQuery x))
+        !gzippedBody = GZip.compress bs
```

Comment:
Should `gzippedBody` and `setContentEncoding` be moved into the else branch? I'm wondering especially about `gzippedBody` if because you've got a strict pattern match that it'll be gzipped even if it won't be used.

Reply:
moved